android/src/main/java/com/pw/droplet/braintree/Braintree.java

```diff
@@ -1,301 +1,316 @@
+            request.setLocaleCode(localeCode);
+            request.setBillingAgreementDescription(description);
+
+            mPayPalClient.requestBillingAgreement(
+                    mCurrentActivity,
+                    request,
+                    e -> handlePayPalResult(null, e));
+        }
 
-    public void nonceErrorCallback(String error) {
-        this.errorCallback.invoke(error);
     }
 
-    private final ActivityEventListener activityEventListener =
-            new BaseActivityEventListener() {
-                @Override
-                public void onActivityResult(Activity activity, int requestCode,
-                                             int resultCode, Intent data) {
-                    int unshiftRequestCode = requestCode & 0x0000ffff;
-                    if (unshiftRequestCode == BraintreeRequestCodes.GOOGLE_PAYMENT) {
-                        if (data != null) {
-                            switch (resultCode) {
-                                case Activity.RESULT_OK:
-                                    GooglePayment.tokenize(mBraintreeFragment,
-                                            PaymentData.getFromIntent(data));
-                                    break;
-                                case Activity.RESULT_CANCELED:
-                                    nonceErrorCallback("USER_CANCELLATION");
-                                    break;
-                            }
-                        } else {
-                            nonceErrorCallback("NO_DATA_AVAILABLE");
-                        }
-                    }
-                }
-            };
+    private void handlePayPalResult(
+            @Nullable PayPalAccountNonce payPalAccountNonce,
+            @Nullable Exception error
+    ) {
+        if (error != null) {
+            handleError(error);
+            return;
+        }
+        if (payPalAccountNonce != null) {
+            sendPaymentMethodNonceResult(payPalAccountNonce.getString());
+        }
+    }
 
     @ReactMethod
-    public void paypalRequest(final Callback successCallback,
-                              final Callback errorCallback) {
-        this.successCallback = successCallback;
-        this.errorCallback = errorCallback;
-        PayPalRequest request = new PayPalRequest().currencyCode("USD").intent(
-                PayPalRequest.INTENT_AUTHORIZE);
-        PayPal.requestBillingAgreement(this.mBraintreeFragment, request);
+    public void showGooglePayViewController(final ReadableMap parameters, final Promise promise) {
+        mPromise = promise;
+
+
+        String currency = "USD";
+        if (!parameters.hasKey("totalPrice")) {
+            promise.reject("You must provide a totalPrice");
+        }
+        if (parameters.hasKey("currencyCode")) {
+            currency = parameters.getString("currencyCode");
+        }
+        if (mCurrentActivity != null) {
+            mGooglePayClient = new GooglePayClient(mBraintreeClient);
```

Comment:
Need to pass in this context
    googlePayClient = new GooglePayClient(mContext, braintreeClient);